### ISSUE TITLE
Loading configuration from mounted files with the same file name only loads the final specified file instead of merging values

### ISSUE THREAD

**Issue by bruce-au** [spring-cloud/spring-cloud-kubernetes#606]
**Describe the bug**
When specifying files to be loaded as configuration via the `spring.cloud.kubernetes.config.paths` option, it only allows for files with unique filenames to be specified. For example, if the following file paths were supplied:

```json
[
"/foo/bar/application.yaml",
"/foo/baz/application.yaml"
]
```

only the configuration of the latter file is loaded into the application.

We've observed this in the v1.1.5.RELEASE.

**Sample**
To reproduce, you can specify two mounted files with the same filename but with different paths under `spring.cloud.kubernetes.config.paths`. Only the second file specified will be loaded into the application.

**Comment by bruce-au** [comment]
@ryanjbaxter @spencergibb ping

**Comment by ryanjbaxter** [comment]
Closed via 4f8ee21
